src/celeritas/ext/detail/ImportProcessConverter.cc

```diff
@@ -637,21 +674,38 @@ ImportProcessConverter::add_micro_xs(G4VEmModel& model)
  * The energy grid is calculated in the same way as in
  * \c G4VEmModel::InitialiseElementSelectors(...) .
  */
-ImportPhysicsVector ImportProcessConverter::initialize_micro_xs_physics_vector(
-    G4VEmModel& model, const ImportMaterial& material)
+ImportPhysicsVector
+ImportProcessConverter::initialize_micro_xs_physics_vector(G4VEmModel&  model,
+                                                           unsigned int mat_id)
 {
+    const auto& material = materials_.at(mat_id);
     CELER_ASSERT(!material.elements.empty());
 
     ImportPhysicsVector physics_vector;
     physics_vector.vector_type = ImportPhysicsVectorType::log;
 
+    double secondary_cutoff = 0;
+    if (process_.secondary_pdg != 0)
+    {
+        auto cutoffs = material.pdg_cutoffs.find(process_.secondary_pdg);
+        CELER_ASSERT(cutoffs != material.pdg_cutoffs.end());
+        secondary_cutoff = cutoffs->second.energy;
+    }
+
     double min_energy = model.LowEnergyLimit() / MeV;
     {
-        auto cutoff_iter = material.pdg_cutoffs.find(process_.particle_pdg);
-        if (cutoff_iter != material.pdg_cutoffs.end())
-        {
-            min_energy = std::max(cutoff_iter->second.energy, min_energy);
-        }
+        const auto& g4_cuts_table
+            = *G4ProductionCutsTable::GetProductionCutsTable();
+        CELER_ASSERT(mat_id < g4_cuts_table.GetTableSize());
+        const auto g4_material
+            = g4_cuts_table.GetMaterialCutsCouple(mat_id)->GetMaterial();
+        const auto g4_particle
+            = G4ParticleTable::GetParticleTable()->FindParticle(
```

Comment:
These are pointer types I think, so better to make it clear that we're not making an unintentional copy:
```suggestion
        const auto* g4_material
            = g4_cuts_table.GetMaterialCutsCouple(mat_id)->GetMaterial();
        const auto* g4_particle
            = G4ParticleTable::GetParticleTable()->FindParticle(
```